### ISSUE TITLE
Shorts next button crashes android browser, both brave and chrome 

### ISSUE THREAD

**Issue by fakeologist** [WWBN/AVideo#8137]
**Describe the bug**

![Screenshot_20230629_155959_Device care.jpg](https://github.com/WWBN/AVideo/assets/66537018/b927e6b3-d1c7-4550-b1ae-e7b85ec05cc7)

https://github.com/WWBN/AVideo/assets/66537018/f0b16279-6e45-4f1a-b6e0-afd9c073ab39

Works ok on Apple mobile Safari browser.

**Comment by GreenDragonTavern** [comment]
@fakeologist - On Android if you open a Shorts video and prior to video finishing, close out, does it blow up browser session?

**Comment by fakeologist** [comment]
No difference although you can't pause and stop a video in shorts

**Comment by GreenDragonTavern** [comment]
Android and AVideo Shorts do not play nice with each other at all.

**Comment by DanielnetoDotCom** [comment]
Ok, I will make more tests on android, now you guys are 2 complainings. 

what exact Android version are you using? what device?

on the devices I have here, it is working fine.

**Comment by GreenDragonTavern** [comment]
> what exact Android version are you using? what device?
Samsung Galaxy S21 - Android v13

**Comment by fakeologist** [comment]
Samsung s20 fe 5g and android 13

**Comment by DanielnetoDotCom** [comment]
Ok, I guess the issue is the Android 13, the android 13 for some reason is not available for the developers so I cannot test it
the latest Android is 12

![image](https://github.com/WWBN/AVideo/assets/18009293/61feef06-45b4-4992-bf08-d7b27b6680a8)

**Comment by DanielnetoDotCom** [comment]
I am installing the Android API 34 (that is the latest available)
I hope I can reproduce it

**Comment by DanielnetoDotCom** [comment]
@GreenDragonTavern are you also using brave?

**Comment by GreenDragonTavern** [comment]
Did a user survey and the phone browsers that came up the most were: Chrome, Firefox, Brave and Edge. I have all installed to verify if there's an issue. I personally use Brave.

**Comment by DanielnetoDotCom** [comment]
ok, so @GreenDragonTavern and @fakeologist are using brave, can you please check if this error happens only on Brave?

**Comment by fakeologist** [comment]
Brave and google chrome

**Comment by GreenDragonTavern** [comment]
Brave, Chrome, Firefox and Edge all fail, so don't think it is browser related.

**Comment by fakeologist** [comment]
It's says a WebView module above what's that anyway ?

**Comment by GreenDragonTavern** [comment]
OK, just verified ***Brave***, ***Edge*** and ***Chrome*** fail. All are built using **Chromium**. Then there's ***Firefox*** and it works flawlessly, no issues. It is the only one of the four not built on **Chromium** platform. Hope this helps you @DanielnetoDotCom.

**Comment by DanielnetoDotCom** [comment]
Yes it helps but looks like the issue is only on the latest Android on Chromium

I will release an option where I will remove the animation, so see if it will stop the crashing. 

currently, I think it is an Android bug, because Chrome on iPhone works fine.

**Comment by DanielnetoDotCom** [comment]
I could not remove the animation, also I think this will not resolve.

Currently, I do not know what to do with the Android 13

**Comment by DanielnetoDotCom** [comment]
I still did not find android 13 to test, but I just tested on android 14 and it works

**Comment by GreenDragonTavern** [comment]
Nothing is ever easy is it 🙃- And thanks for update.

Quick question, isn't there a more eloquent way to identify Shorts besides duration? Wouldn't using the ***type*** field in the end give you more flexibility in 1) identifying, 2) restrictions ie; proper format, duration, 3) Encoder could automatically identify based on Short format and 4) Coding for Shorts is consistent based on specs and not just how long a video lasts.

**Comment by fakeologist** [comment]
Thanks for the effort. I am guessing most people have 13, I doubt my phone can update to 14. Good excuse to get a new one.

**Comment by DanielnetoDotCom** [comment]
I still could not test on android 13, but it could also be something on Samsung

**Comment by DanielnetoDotCom** [comment]
I could test on Android 13 and it works, so I can only suspect it is a Samsung issue

**Comment by DanielnetoDotCom** [comment]
this update will update some libraries, please check if fixes the Samsung issue.

**Comment by fakeologist** [comment]
how the shorts title shows up without shorts at all on android.
not playing on windows 11 desktop either.

**Comment by DanielnetoDotCom** [comment]
> how the shorts title shows up without shorts at all on android. not playing on windows 11 desktop either.

Sorry what?

**Comment by fakeologist** [comment]
No videos display 
![Screenshot_20230630_081355_Brave.jpg](https://github.com/WWBN/AVideo/assets/66537018/f53e4a5b-f185-43c8-9983-5f295824bd4b)

**Comment by fakeologist** [comment]
also shorts don't autoplay on desktop and there is no play button in the middle of the screen to start video.

**Comment by DanielnetoDotCom** [comment]
I see, check the gallery plugin and disable shorts if you do not want

I will send an update later regarding it

**Comment by DanielnetoDotCom** [comment]
> also shorts don't autoplay on desktop and there is no play button in the middle of the screen to start video.

autoplay is tricky, I cannot guarantee, check this https://github.com/WWBN/AVideo/wiki/Autoplay-and-Browser-Policies

but the play button should be visible. can you give me a sample where you cannot see the button?

**Comment by GreenDragonTavern** [comment]
Shorts interface is sweet on Firefox. Brave, Chrome and Edge still nukes playing Shorts. BTW, Samsung Galaxy uses One UI interface and here is link to emulators: https://developer.samsung.com/galaxy-emulator-skin

**Comment by fakeologist** [comment]
Autoplay worked fine before your changes.
Shorts will be great once you figure out the issue.

**Comment by DanielnetoDotCom** [comment]
I sent you updates, please try now

**Comment by fakeologist** [comment]
Doesn't crash the browse when scrolling, but also won't play.

Regular videos have no controls on them either, including progress bar

**Comment by DanielnetoDotCom** [comment]
please make sure you clear your browser cache. I just tested and you are fine now

**Comment by DanielnetoDotCom** [comment]
![chrome-capture-2023-5-30](https://github.com/WWBN/AVideo/assets/18009293/10e9970f-19c8-488d-a26a-f47ec8127ac4)

**Comment by fakeologist** [comment]
> ![chrome-capture-2023-5-30](https://github.com/WWBN/AVideo/assets/18009293/10e9970f-19c8-488d-a26a-f47ec8127ac4)
> 

It's back to crashing after advance to next video

**Comment by DanielnetoDotCom** [comment]
I did not see the crash

**Comment by fakeologist** [comment]
> I did not see the crash 

It's back to the way it was at the start of this thread. Next video crashes the browser.

**Comment by DanielnetoDotCom** [comment]
Looks like the crash only happens on Samsung. 

I could not reproduce it.

**Comment by GreenDragonTavern** [comment]
> Looks like the crash only happens on Samsung.
> 
> I could not reproduce it.

https://developer.samsung.com/galaxy-emulator-skin

**Comment by fakeologist** [comment]
> Looks like the crash only happens on Samsung. 
> 
> I could not reproduce it. 

Too bad because my next phone will be a Samsung as I can't find many of the apps I use daily on iOS.

**Comment by fakeologist** [comment]
Update: shorts works with Firefox on android 13 Samsung 20 fe 5g

**Comment by DanielnetoDotCom** [comment]
> > Looks like the crash only happens on Samsung.
> > I could not reproduce it.
> 
> https://developer.samsung.com/galaxy-emulator-skin

the skin does not help, it is about the hardware "I guess";

maybe in a next Samsung update it will be fixed

**Comment by fakeologist** [comment]
> > > Looks like the crash only happens on Samsung.
> > > I could not reproduce it.
> > 
> > 
> > https://developer.samsung.com/galaxy-emulator-skin
> 
> the skin does not help, it is about the hardware "I guess";
> 
> maybe in a next Samsung update it will be fixed

If it works in firefox how can it be a Samsung issue?

**Comment by DanielnetoDotCom** [comment]
Because this is only happening on Samsung. I haven't encountered this issue in the devices I have available here. also I tested on Android 13 without any problem

**Comment by GreenDragonTavern** [comment]
No biggie, it's not like Samsung dominates the Android market or something like that 🙃

**Comment by GreenDragonTavern** [comment]
Daniel has gotten plenty of kudos here but when I read "maybe in a next Samsung update it will be fixed". One must ponder, did Samsung acknowledge their AVideo bug?

**Comment by JoshWho** [comment]
It happens on Motorola also. My phone crashes in vertical view. Wide screen does not but vertical crashes when you click the next button. PWA and browser. Also the Like button is blocking the next button in wide view.

**Comment by DanielnetoDotCom** [comment]
I really need to be able to reproduce it, on my side it is not happening.

Maybe you guys can help me to find out

try this for js errors
https://developer.chrome.com/docs/devtools/remote-debugging/

also, you can try to debug your Android device with android studio, to get hardware and OS errors

**Comment by JoshWho** [comment]
Are you holding your phone vertical when doing it?

**Comment by DanielnetoDotCom** [comment]
I did on both positions

**Comment by JoshWho** [comment]
When I click the next arrow it makes the screen show the next video thumbnail but slides halfway then it crashes

**Comment by JoshWho** [comment]
Autoplay it does not crash, just when I click it

**Comment by JoshWho** [comment]
Whatever the animation is before going to the next video is what is causing the crash. Seems like a buffer overrun crashes the browser app, probably a memory leak

**Comment by JoshWho** [comment]
https://github.com/WWBN/AVideo/issues/8159
